pkg/controllers/nodepool/hash/controller.go

```diff
@@ -69,3 +89,30 @@ func (c *Controller) Builder(_ context.Context, m manager.Manager) operatorcontr
 		WithOptions(controller.Options{MaxConcurrentReconciles: 10}),
 	)
 }
+
+func (c *Controller) updateNodeClaimHash(ctx context.Context, np *v1beta1.NodePool, nphash string) error {
+	ncList := &v1beta1.NodeClaimList{}
+	if err := c.kubeClient.List(ctx, ncList); err != nil {
```

Comment:
Can we consider creating an indexer here for the hash version and then using a field selector to get all the NodeClaims that don't match the current version. This is going to save on memory and performance